hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieGlobalTimeline.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.table.timeline;
+
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.ValidationUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * A global timeline view with both active and archived timeline involved.
+ */
+public class HoodieGlobalTimeline extends HoodieDefaultTimeline {
+  private static final long serialVersionUID = 1L;
+  private static final Logger LOG = LoggerFactory.getLogger(HoodieGlobalTimeline.class);
+  private final HoodieTableMetaClient metaClient;
+  private final HoodieActiveTimeline activeTimeline;
+  private final HoodieArchivedTimeline archivedTimeline;
+
+  protected HoodieGlobalTimeline(HoodieTableMetaClient metaClient, Option<String> startInstant) {
+    this.metaClient = metaClient;
+    this.activeTimeline = new HoodieActiveTimeline(metaClient);
+    archivedTimeline = startInstant.isPresent() ? new HoodieArchivedTimeline(metaClient, startInstant.get()) : new HoodieArchivedTimeline(metaClient);
+    this.details = FederatedDetails.create(this.activeTimeline, archivedTimeline);
+    setInstants(mergeInstants(archivedTimeline.getInstants(), activeTimeline.getInstants()));
```

Comment:
are we leading all the archived instants here eagerly? can this be lazily done?

Reply:
Yeah, that could be more smart, in this PR, when the `startTs` is empty, the archived timeline loading actually does nothing, but I think we can even make the loading lazy when `startTs` instant itself is active.

Reply:
I kind of make the archived timeline loading smartly based on the given `startTs`, if we also wanna lazy loading for specific API invocation level, that might needs a refactoring of all the timelines to support lazy setup of the instants.

Would prefer to to that in another PR if we reach consensus.